### PR TITLE
api,server,ui: add project ID and name to UserDataResponse

### LINKED ISSUE #8363: Add project information to UserData Response
Currently when create a userdata in a project, it looks like
image

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.18/4.19
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #8363 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

image

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/response/UserDataResponse.java [hunk 1]
```diff
@@ -40,6 +40,14 @@ public class UserDataResponse extends BaseResponseWithAnnotations {
     @SerializedName(ApiConstants.ACCOUNT) @Param(description="the owner of the userdata")
     private String accountName;
 
+    @SerializedName(ApiConstants.PROJECT_ID)
+    @Param(description = "the project id of the userdata")
+    private String projectId;
+
+    @SerializedName(ApiConstants.PROJECT)
+    @Param(description = "the project name of the userdata")
```
Comment (sureshanaparti) [inline api/src/main/java/org/apache/cloudstack/api/response/UserDataResponse.java:48]: _since_ attribute needed for both params?

**Comment by DaanHoogland** [comment]
@weizhouapache , given the name of your branch is this for 4.19?

**Comment by weizhouapache** [comment]
> @weizhouapache , given the name of your branch is this for 4.19?

@DaanHoogland 
the branch is based on 4.19.
However, I added to 4.20.0.0 milestone as I am not sure if it is a bug fix. If it is, we can change the target branch to 4.19

**Comment by DaanHoogland** [comment]
I think a backwards compatible enhancement like this can go into a minor release.

**Comment by weizhouapache** [comment]
> I think a backwards compatible enhancement like this can go into a minor release.

OK @DaanHoogland 
changed the target branch to 4.19
I will add `since` in response.

**Comment by weizhouapache** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/response/UserDataResponse.java (hunk 1 above)
Comment (weizhouapache) [reply api/src/main/java/org/apache/cloudstack/api/response/UserDataResponse.java]: yes,added.
thanks @sureshanaparti

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Comment by DaanHoogland** [comment]
@sureshanaparti , I think we can merge at your will.

**Review by sureshanaparti — APPROVED** [review]
LGTM

**Pull request merged**